src/basilisp/contrib/bencode.lpy

```diff
@@ -0,0 +1,132 @@
+(ns basilisp.contrib.bencode
+  "A port of nbb's bencode implementation to basilisp.
+
+  https://github.com/babashka/nbb/blob/bca8b5017a06768eb35d02a2d6233ca9c6c2f692/src/nbb/impl/bencode.cljs")
+
+(defn- index-of [s c]
+  (let [i (.index s c)]
+    (if (< i 0) (throw (python/EOFError "out of input")) i)))
+
+(defn- slice
+  ([buffer start]
+   (if (< (len buffer) start)
+     (throw (python/EOFError "out of input"))
+     (.__getitem__ buffer (python/slice start nil))))
+  ([buffer start end]
+   (if (> end (len buffer))
+     (throw (python/EOFError "out of input"))
+     (.__getitem__ buffer (python/slice start end)))))
```

Comment:
Are slices `get`-able too? If not that could be another area we could improve the API of `get`.

Reply:
`get` worked fine in the above, again a remnant of early attempts to use bytes directly.

The upgraded interface for `slice` now works bytes, and I've added the corresponded decorator in the runtime to support this..